### PR TITLE
FocusObjectChanged if focused object gets destroyed

### PR DESCRIPTION
**Pull request by janholo**
Added a condition to send the FocusObjectChanged event if the focused object get destroyed

### CODE REVIEW

File: Assets/HoloToolkit/Input/Scripts/Gaze/GazeManager.cs [hunk 1]
```diff
@@ -130,6 +130,7 @@ private void Update()
             UpdateGazeInfo();
 
             // Perform raycast to determine gazed object
+            bool isPreviousGazingAtObject = IsGazingAtObject;
```
Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/Gaze/GazeManager.cs]: What changes this to something other than false?

**Comment by HodgsonSDAS** [comment]
Sorry, I dislike how github folds important stuff when reviewing haha. I understand what's going on now.

We set the isPreviousGazingAtObject to our IsGazingAtObject value, perform a raycast, then check the values.

**Comment by HodgsonSDAS** [comment]
I think this change will also affect how `RaycastUnityUI()` works as well. 

We set `IsGazingAtObject` [true when we're either not gazing at an object is between us and the UI](https://github.com/janholo/HoloToolkit-Unity/blob/e2e0b7d6b6e1791840b0ee3d84728610e7daec7e/Assets/HoloToolkit/Input/Scripts/Gaze/GazeManager.cs#L269).

This will change how we evaluate the need to throw a `FocusedObjectChanged` event.

**Comment by StephenHodgson** [comment]
Closed do to inactivity.

**Pull request closed without merging**